pliers/extractors/models.py

```diff
@@ -1,14 +1,193 @@
 ''' Extractor classes based on pre-trained models. '''
 
 import numpy as np
+import pandas as pd
 
 from pliers.extractors.image import ImageExtractor
-from pliers.extractors.base import ExtractorResult
+from pliers.extractors.base import Extractor, ExtractorResult
 from pliers.filters.image import ImageResizingFilter
-from pliers.utils import attempt_to_import, verify_dependencies
+from pliers.stimuli import ImageStim, TextStim, AudioStim
+from pliers.utils import (attempt_to_import, verify_dependencies,
+                         listify)
+
+import logging
 
 tf = attempt_to_import('tensorflow')
+hub = attempt_to_import('tensorflow_hub')
 attempt_to_import('tensorflow.keras')
+attempt_to_import('tensorflow_text')
+
+
+class TFHubExtractor(Extractor):
+
+    ''' A generic class for Tensorflow Hub extractors 
+    Args:
+        url_or_path (str): url or path to TFHub model. You can
+            browse models at https://tfhub.dev/.
+        task (str): model task/domain identifier
+        features (optional): list of labels (for classification) 
+            or other feature names. If a one-element list or 
+            a string is passed, packs output vectors into one column.
+            If not specified, returns numbered features 
+            (feature_0, feature_1, ... ,feature_n).
+        kwargs (dict): arguments to hub.KerasLayer call
+    '''
+
+    _log_attributes = ('url_or_path', 'features', 'transform_out')
```

Comment:
This is one of the problems with using generic classes: we lose the ability to easily log potentially important initialization parameters. I don't know if there's an easy solution to this... if `kwargs` can only contain fairly simple, hashable types, we could just add that to `_log_attributes`. But I'm guessing it's not that predictable, in which case I guess we just have to live with it.

Reply:
indeed, arguments to `KerasLayer ` can be pretty much anything depending on the model, and some of it is non-hashable (e.g. the "arguments" dictionary: https://www.tensorflow.org/hub/api_docs/python/hub/KerasLayer). Would lean towards leaving it as it is, but I can add a warning if kwargs are passed if needed.